File: AddresDotNet/IBuilder.cs
namespace AddresDotNet;
/// <summary>
/// Интерфейс для объекта Строитель.
/// </summary>
public interface IBuilder
{
    /// <summary>
    /// Собирает и возвращает объект.
    /// </summary>
    /// <returns>Готовый объект.</returns>
    Address Build();
}

File: AddressKotlin/src/main/kotlin/interfaces/IBuilder.kt
package interfaces

import model.Address

/**
 * Контракт для строителей объектов адреса.
 */
interface IBuilder {

    /**
     * Собирает и возвращает итоговый объект адреса.
     *
     * @return Сконструированный объект [Address].
     */
    fun build(): Address
}